### ISSUE TITLE
[MacOS ARM] Simpsons Game Menu Doesn't Work

### ISSUE THREAD

**Issue by Icnopandaz** [RPCS3/rpcs3#16424]
### Quick summary

Inputs in 2D menu work for everything BUT up/down (Can only click top prompt)

### Details

Inputs work for everything BUT the menus.

Means after 2nd mission when needing to play as Bart/Lisa you're softlocked.

I went around this by grabbing the X64 MacOS Version, where the menu DOES work can go up/down to select options. This is important as to proceed with missions you need to use the menu to change characters, are softlocked on the ARM Mac version after the first 2 missions. 

No issues outside of the 2D Menu. 

### Attach a log file

Its purely button inputs, was told in Discord I shouldn't need to attach logs. Will do however if needed let me know lets get to the bottom of this!

### System configuration

Macbook M1 Air - 8GB / 256GB.

**Comment by digant73** [comment]
@Icnopandaz is that still an issue?

**Comment by Icnopandaz** [comment]
> [@Icnopandaz](https://github.com/Icnopandaz) is that still an issue?

Have not tested since, I swapped to the Xbox360 emulator if I recall correctly though never 100%'d in the end.

**Comment by DiegoHSO** [comment]
Hi, this is still an issue here on a MacBook Pro - 24GB/512GB - M4 Pro chip. Using both DualSense and keyboard inputs, I'm unable to navigate through the menus.

**Comment by digant73** [comment]
@DiegoHSO please an update on current build

**Comment by DiegoHSO** [comment]
Hi @digant73, the issue keeps happening for both DualSense and keyboard inputs on build 0.0.41-19499 for macOS arm64 (released June 19th, 2026).

**Comment by Icnopandaz** [comment]
Hi @digant73 @DiegoHSO 

Have some potentially good news. Obviously, I posted this over a year and a half ago now, but seeing more notifications I decided to be nosey into the code..

Bug: MacOS ARM treats arrow keys differently than X64, and doesn't work due to a name mismatch in keyboard_pad_handler.cpp (rpcs3/Input/keyboard_pad_handler.cpp)

GetKeyName() Line 883-891 returns Unicode arrow symbols, Line 944-950 looks it up, but only handles Num versions, not symbols

Fix - Add the plain arrow symboll WITH the Num prefixed ones.


Lines 946 or so (Forget exact number)

if (keyName == "Num←" || keyName == "←") return Qt::Key_Left;
if (keyName == "Num↑" || keyName == "↑") return Qt::Key_Up;
if (keyName == "Num→" || keyName == "→") return Qt::Key_Right;
if (keyName == "Num↓" || keyName == "↓") return Qt::Key_Down;
#endif 

TLDR: QKeySequence doesn't work properly for arrow keys on ARM MacOS. Amusingly, its literally a code comment where the bug is, just whomever put the comment there never bothered to fix it?

Link to Pull Request: https://github.com/RPCS3/rpcs3/pull/18920

**Comment by kd-11** [comment]
But the problem is stated as affecting controllers as well.

**Comment by digant73** [comment]
@Icnopandaz if this issue is fixed now that your PR has been merged, close this ticket please

**Comment by DiegoHSO** [comment]
Hi guys, the issue is still happening for me in v0.0.41-19558 macOS arm64 build – both keyboard and DualSense inputs. Can anyone else confirm if the fix didn't work as well?

**Comment by digant73** [comment]
@Icnopandaz please provide an update on current build

**Comment by Chokolite** [comment]
Have the same problem on macbook pro m1 pro 16gb with dualsens or keyboard. Canary fix did not help.
RPCS3 Version: 0.0.42-19884-3ef20ebb Alpha | master
Game - BLES00142
